RoaringBitmap/src/main/java/org/roaringbitmap/longlong/LongUtils.java

```diff
@@ -93,31 +93,35 @@ public static long fromBDBytes(byte[] work) {
    *
    * @param a a byte array
    * @param b another byte array
-   * @return 1 indicates a greater than b,0 indicates equal,-1 indicates a smaller than b
+   * @return positive indicates a greater than b, 0 indicates equal,negative otherwise
    */
   public static int compareHigh(byte[] a, byte[] b) {
-    return compareTo(a, 0, a.length, b, 0, b.length);
+    if (a == b) { return 0; }
+    if (a[0] != b[0]) { return Byte.toUnsignedInt(a[0]) - Byte.toUnsignedInt(b[0]); }
+    if (a[1] != b[1]) { return Byte.toUnsignedInt(a[1]) - Byte.toUnsignedInt(b[1]); }
+    if (a[2] != b[2]) { return Byte.toUnsignedInt(a[2]) - Byte.toUnsignedInt(b[2]); }
+    if (a[3] != b[3]) { return Byte.toUnsignedInt(a[3]) - Byte.toUnsignedInt(b[3]); }
+    if (a[4] != b[4]) { return Byte.toUnsignedInt(a[4]) - Byte.toUnsignedInt(b[4]); }
+    if (a[5] != b[5]) { return Byte.toUnsignedInt(a[5]) - Byte.toUnsignedInt(b[5]); }
+    return 0;
   }
 
-  private static int compareTo(byte[] buffer1, int offset1, int length1,
-      byte[] buffer2, int offset2, int length2) {
-    if (buffer1 == buffer2
-        && offset1 == offset2
-        && length1 == length2) {
-      return 0;
-    }
-    int end1 = offset1 + length1;
-    int end2 = offset2 + length2;
-    for (int i = offset1, j = offset2; i < end1 && j < end2; i++, j++) {
-      int a = (buffer1[i] & 0xff);
-      int b = (buffer2[j] & 0xff);
-      if (a != b) {
-        return a - b;
-      }
-    }
-    return length1 - length2;
+  /**
+   * Equality test according to the dictionary order.
+   *
+   * @param a a byte array
+   * @param b another byte array
+   * @return true if first six bytes of both given array are equal
+   */
+  public static boolean equalsHigh(byte[] a, byte[] b) {
+    return (a == b)
```

Comment:
If a==b is common then it is worth having, but if it is uncommon, then it could be a wasted effort.

How often is a==b in practice?

Reply:
You are faster than light, sorry for my incautiousness as I see now some failing tests, do not waste your time before I will check this.

Reply:
Please be patient as I need to take a rest as work from early morning I will look at this as soon as possible. My MR are not so important except one issue not being already MR - if somebody is boring -  [SYSTEM_ARRAYCOPY](https://github.com/xtonik/RoaringBitmap/tree/SYSTEM_ARRAYCOPY), however I broke something and I slowly fixing it.